lambda/marchon.py

```diff
@@ -76,32 +94,62 @@ def get_sheet_data():
      16 Event Link, 17 Photo, 18 About
     '''
     # keep these fields
-    fields = {'name': 0, 'location': 1, 'contactName': 3, 'contactEmail': 5,
-              'facebook': 9, 'twitter': 10, 'instagram': 11, 'other': 12,
-              'website': 13, 'event': 14, 'eventDate': 15, 'eventLink': 16,
-              'photo': 17, 'about': 18}
+    fields = {
+        'name': 0,
+        'location': 1,
+        'contactName': 3,
+        'contactEmail': 5,
+        'facebook': 9,
+        'twitter': 10,
+        'instagram': 11,
+        'other': 12,
+        'website': 13,
+        'event': 14,
+        'eventDate': 15,
+        'eventLink': 16,
+        'photo': 17,
+        'about': 18
+    }
     return read_sheet('Sheet1!A1:S', fields, 1, True)
 
 
+def get_action_network_events() -> Dict:
+    log.info('\nstart get Action Network events')
+    return_events = get_events_from_events_campaign()
+    log.info('\ngot %d events from action network', len(return_events))
+    return return_events
+
+
 def get_geojson(url):
     log.info('\nload geojson')
     resp = requests.get('https://s3.amazonaws.com/ragtag-marchon/%s' % url)
     features = {}
     for feature in resp.json()['features']:
-        features[feature['properties']['location']] = feature
+        if feature['properties'].get('source', '') == 'actionnetwork':
+            key = make_key(feature)
+        else:
+            key = feature['properties']['location']
+        features[key] = feature
     log.info('read %s features', len(features))
     return features
 
 
+def get_location_from_key(key: str) -> str:
+    parts = key.split('::', 1)
+    return parts[0]
+
```

Comment:
why is this necessary?

Reply:
1. You said zip code is a sufficient location for house parties on this map
2. You said zip code works reliably for geocoding in map box
3. To point 2, `New York, NY 10025` failed to geocode while `NY, NY 10025` did geocode, so I went with zip code only because it's reliable and accurate enough
4. There could be more than one party in a given zip code